src/twisted/conch/ssh/keys.py

```diff
@@ -311,14 +320,100 @@ def _fromString_PUBLIC_OPENSSH(cls, data):
         blob = decodebytes(data.split()[1])
         return cls._fromString_BLOB(blob)
 
+
     @classmethod
-    def _fromString_PRIVATE_OPENSSH(cls, data, passphrase):
+    def _fromPrivateOpenSSH_v1(cls, data, passphrase):
         """
         Return a private key object corresponding to this OpenSSH private key
-        string.  If the key is encrypted, passphrase MUST be provided.
-        Providing a passphrase for an unencrypted key is an error.
+        string, in the "openssh-key-v1" format introduced in OpenSSH 6.5.
+
+        The format of an openssh-key-v1 private key string is::
+            -----BEGIN OPENSSH PRIVATE KEY-----
+            <base64-encoded SSH protocol string>
+            -----END OPENSSH PRIVATE KEY-----
+
+        The SSH protocol string is as described in
+        U{PROTOCOL.key<https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/usr.bin/ssh/PROTOCOL.key>}.
+
+        @type data: L{bytes}
+        @param data: The key data.
+
+        @type passphrase: L{bytes} or L{None}
+        @param passphrase: The passphrase the key is encrypted with, or L{None}
+        if it is not encrypted.
+
+        @return: A new key.
+        @rtype: L{twisted.conch.ssh.keys.Key}
+        @raises BadKeyError: if
+            * a passphrase is provided for an unencrypted key
+            * the SSH protocol encoding is incorrect
+        @raises EncryptedKeyError: if
+            * a passphrase is not provided for an encrypted key
+        """
+        lines = data.strip().splitlines()
+        keyList = decodebytes(b''.join(lines[1:-1]))
+        if not keyList.startswith(b'openssh-key-v1\0'):
+            raise BadKeyError('invalid new-format OpenSSH private key')
+        keyList = keyList[len(b'openssh-key-v1\0'):]
+        cipher, rest = common.getNS(keyList)
+        kdf, rest = common.getNS(rest)
+        kdfOptions, rest = common.getNS(rest)
+        n = struct.unpack('!L', rest[:4])[0]
```

Comment:
Is there a reason not to use getNS with a count argument here? (I'm not suggesting I'd insist on the change, it just looked strange to me.)

Reply:
Ah, of course, sorry :)

Reply:
I might be misunderstanding your response, but wouldn't `cipher, kdf, kdfOptions, rest = common.getNS(keyList, count=3)` do the right thing here? (Again, not a big deal, but since lvh pointed it out…)

Reply:
Oh, I see, maybe @lvh's comment was just one line below where it should have been?  I get it now; fixed.